core/cap-0063.md

```diff
@@ -0,0 +1,496 @@
++    int64 feeReadLiveSoroban1KB;    // Fee for reading 1KB of live Soroban state
+ 
+     // The following parameters determine the write fee per 1KB.
+     // Write fee grows linearly until bucket list reaches this size
+@@ -60,6 +60,26 @@ struct ConfigSettingContractLedgerCostV0
+     uint32 bucketListWriteFeeGrowthFactor;
+ };
+ 
++struct ConfigSettingContractLedgerCostExtV1
++{
++    // Maximum number of evicted Soroban and classic entry read operations per ledger
++    uint32 ledgerMaxReadEvictedAndClassicEntries;
++
++    // Maximum number of bytes of evicted Soroban and classic state that can be
++    // read per ledger
++    uint32 ledgerMaxReadEvictedAndClassicBytes;
++
++    // Maximum number of evicted Soroban and classic entry read operations per transaction
++    uint32 txMaxReadEvictedAndClassicEntries;
++
++    // Maximum number of bytes of evicted Soroban and classic state that can be
++    // read per transaction
++    uint32 txMaxReadEvictedAndClassicBytes;
++
++    int64 feeReadEvictedAndClassicEntry;  // Fee per evicted Soroban/classic entry read
++    int64 feeReadEvictedAndClassic1KB;    // Fee for reading 1 KB of evicted Soroban/classic state
++};
++
+ // Historical data (pushed to core archives) settings for contracts.
+ struct ConfigSettingContractHistoricalDataV0
+ {
+@@ -302,7 +322,8 @@ enum ConfigSettingID
+     CONFIG_SETTING_STATE_ARCHIVAL = 10,
+     CONFIG_SETTING_CONTRACT_EXECUTION_LANES = 11,
+     CONFIG_SETTING_BUCKETLIST_SIZE_WINDOW = 12,
+-    CONFIG_SETTING_EVICTION_ITERATOR = 13
++    CONFIG_SETTING_EVICTION_ITERATOR = 13,
++    CONFIG_SETTING_CONTRACT_LEDGER_COST_EXT_V1 = 14
+ };
+ 
+ union ConfigSettingEntry switch (ConfigSettingID configSettingID)
+@@ -335,5 +356,7 @@ case CONFIG_SETTING_BUCKETLIST_SIZE_WINDOW:
+     uint64 bucketListSizeWindow<>;
+ case CONFIG_SETTING_EVICTION_ITERATOR:
+     EvictionIterator evictionIterator;
++case CONFIG_SETTING_CONTRACT_LEDGER_COST_EXT_V1:
++    ConfigSettingContractLedgerCostExtV1 ledgerCostExtV1;
+ };
+ }
+diff --git a/Stellar-transaction.x b/Stellar-transaction.x
+index 7d32481..0abe3e4 100644
+--- a/Stellar-transaction.x
++++ b/Stellar-transaction.x
+@@ -888,10 +888,31 @@ struct SorobanResources
+     uint32 writeBytes;
+ };
+ 
++struct SorobanResourcesExtV1
++{
++    // The maximum number of bytes of evicted Soroban state
++    // and classic state that can be read
++    uint32 readEvictedAndClassicBytes;
++
++    // Bitmap representing what Soroban entries in the
++    // footprint are evicted, based on the order of keys
++    // provided in the LedgerFootprint vectors. readOnly
++    // vector is first, followed by readWrite.
++    // If no keys are evicted, BitMap can be empty.
++    BitMap evictedSorobanEntries;
++};
++
+ // The transaction extension for Soroban.
+ struct SorobanTransactionData
+ {
+-    ExtensionPoint ext;
++    union switch (int v)
++    {
++    case 0:
++        void;
++    case 1:
++        SorobanResourcesExtV1 resourceExt;
```

Comment:
What about the proofs, do we allocate an ext for these?

Reply:
I didn't include it in this CAP just because they aren't necessary until full state archival.